python/tank/deploy/tank_commands/get_entity_commands.py

```diff
@@ -0,0 +1,274 @@
+                          "Details: %s"
+                          % (pipeline_config_path, entity_type, e))
+
+        return commands_per_entity
+
+    def _get_cache_name(self, platform, entity_type):
+        """
+        Constructs the expected name for the cache file of a particular entity
+        type.
+
+        :param platform:    platform that will use the cached information
+        :param entity_type: entity type that we want the cache for
+        :returns:           name of the file containing the desired cached data
+        """
+        # get a platform name that follows the conventions of the shotgun cache
+        platform_name = platform
+        if platform == "darwin":
+            platform_name = "mac"
+        elif platform == "win32":
+            platform_name = "windows"
+        elif platform.startswith("linux"):
+            platform_name = "linux"
+
+        return ("shotgun_%s_%s.txt" % (platform_name, entity_type)).lower()
+
+    def _get_env_name(self, entity_type):
+        """
+        Constructs the expected name for the environment file of a particular
+        entity type. This environment file should contain the shotgun engine
+        with the apps that will register the desired commands.
+
+        :param entity_type: entity type that we want the environment for
+        :returns:           name of the file with the desired environment
+        """
+        return "shotgun_%s.yml" % entity_type.lower()
+
+    def _load_cached_data(self, pipeline_config_path, entity_type):
+        """
+        Loads the cached data for the given entities from the specified
+        Pipeline Configuration.
+
+        This is done by invoking the toolkit command of the other Pipeline
+        Configuration to update the cache (if needed) and get the cache
+        content.
+
+        :raises:                     will raise a TankError if we were not able
+                                     to update the cache or get its content
+        :param pipeline_config_path: path to the Pipeline Configuration
+                                     containing the cache that we want
+        :param entity_type:          type of the entity we want the cache for
+        :returns:                    text data contained in the cache
+        """
+        cache_name = self._get_cache_name(sys.platform, entity_type)
+        env_name = self._get_env_name(entity_type)
+
+        # try to load the data right away if it is already cached
+        try:
+            return execute_toolkit_command(pipeline_config_path,
+                                           "shotgun_get_actions",
+                                           [cache_name, env_name])
+        except SubprocessCalledProcessError, e:
+            # failed to load from cache - only OK if cache is missing or out
+            # of date
+            if e.returncode not in [self._ERROR_CODE_CACHE_OUT_OF_DATE,
+                                    self._ERROR_CODE_CACHE_NOT_FOUND]:
```

Comment:
sorry, i see this now - please ignore comment above about the `range(2)` stuff.  :)